mono/mini/mini-llvm.c

```diff
@@ -3579,7 +3579,10 @@ emit_entry_bb (EmitContext *ctx, LLVMBuilderRef builder)
 			ctx->addresses [reg] = LLVMGetParam (ctx->lmethod, pindex);
 			break;
 		default:
-			ctx->values [reg] = convert_full (ctx, ctx->values [reg], llvm_type_to_stack_type (cfg, type_to_llvm_type (ctx, ainfo->type)), type_is_unsigned (ctx, ainfo->type));
+			if (ainfo->type->byref)
+				ctx->values [reg] = convert_full (ctx, ctx->values [reg], IntPtrType(), FALSE);
+			else
+				ctx->values [reg] = convert_full (ctx, ctx->values [reg], llvm_type_to_stack_type (cfg, type_to_llvm_type (ctx, ainfo->type)), type_is_unsigned (ctx, ainfo->type));
```

Comment:
Match the logic here:

https://github.com/mono/mono/blob/9608debd7bdf94fe27258d7365d7c04d1c648d90/mono/mini/ssa.c#L440-L443